apps/hash-api/src/graphql/create-apollo-server.ts

```diff
@@ -64,7 +64,16 @@ export const createApolloServer = ({
     context: (ctx): Omit<GraphQLContext, "dataSources"> => ({
       ...ctx,
       authentication: {
-        actorId: ctx.req.user?.accountId ?? publicUserAccountId,
+        /**
+         * Users who haven't completed signup are only permitted to set a name and shortname.
+         * For all other purposes they are treated as an anonymous user, and so we default them to one here.
+         *
+         * We have middleware which also checks this, but this default ensures that forgetting to apply
+         * the middleware to a new route doesn't give an incomplete user permissions they shouldn't have.
+         */
+        actorId: ctx.req.user?.isAccountSignupComplete
+          ? ctx.req.user.accountId
+          : publicUserAccountId,
```

Comment:
I'm not sure we about enforcing this here as we need to remember in the resolvers that the `actorId` might be unreliable if we are dealing with users who aren't signed up. I think I'd prefer just using the middleware check in the BE.

Reply:
I've updated how we handle this in https://github.com/hashintel/hash/pull/3380/commits/5582e75ac1ba815b11ae6149252646e999de37ae following @TimDiekmann's work in #3408 – this allows us to ensure users don't accidentally get permission to do things that they shouldn't (since we don't create them a web until they complete signup) while maintaining their proper actor id for all requests

Reply:
I had to update the tests in https://github.com/hashintel/hash/pull/3380/commits/6d84b5da16437321e36b4ad76f04ce503294c132, because they weren't setting the user's `preferredName` – good example of where the new system is catching things the old one didn't!